profiler/data_collector.rst

```diff
@@ -73,6 +78,11 @@ The getters are added to give the template access to the collected information.
 
 .. caution::
 
+    If the data that is not directly related to the request or response, you need to make the data accessible to your DataCollector.
+    This can be achieved by injecting the service that holds the information you intend to profile into your DataCollector.
```

Comment:
I think that's a bit too specific. But let's see what other members of the @symfony/team-symfony-docs team think about it.